content/blaze.md

```diff
@@ -0,0 +1,543 @@
+---
+title: Blaze
+---
+
+After reading this guide, you'll know:
+
+1. How to use the Spacebars syntax, used to define Blaze templates.
+2. Best practice towards creating reusable components in Blaze.
+3. How the Blaze rendering engine works under the hood and some techniques to best use it.
+4. How to test Blaze templates.
+
+Blaze is Meteor's built in reactive rendering library. Using templates written in [Spacebars](https://github.com/meteor/meteor/blob/devel/packages/spacebars/README.md), a variant of [Handlebars](http://handlebarsjs.com) designed to take advantage of [Tracker](https://github.com/meteor/meteor/tree/devel/packages/tracker), Meteor's reactivity system.
+
+Blaze is not required to built applications in Meteor---you can equally use [React](http://react-in-meteor.readthedocs.org/en/latest/) or [Angular](http://www.angular-meteor.com), however this guide will take you through best practice in building an application in Blaze.
+
+<a id="spacebars">Spacebars</a>
+
+Spacebars is a handlebars-like templating language, built off the concept of rendering a reactivily changing *data context*. Spacebars templates look like simple HTML with special "mustache" tags delimited by `{{` and `}}`.
+
+As an example, consider the `todosItem` template from the Todos example app:
+
+```html
+<template name="todosItem">
+  <div class="list-item {{checkedClass}} {{editingClass}}">
+    <label class="checkbox">
+      <input type="checkbox" checked={{todo.checked}} name="checked">
+      <span class="checkbox-custom"></span>
+    </label>
+
+    <input type="text" value="{{todo.text}}" placeholder="Task name">
+    <a class="js-delete-item delete-item" href="#">
+      <span class="icon-trash"></span>
+    </a>
+  </div>
+</template>
+```
+
+In this example, the template is rendered with an object with key `todo` as data context (we'll see below how to enforce that). We access the properties of the `todo`  using the mustache tag, such as `{{todo.text}}`. The default behaviour is to render that property as a string; however in some cases (such as `checked=={{todo.checked}}` it can be resolved as a boolean value).
+
+Note that simple string interpolations like this will never render HTML---so you don't need to perform safety checks for XSS.
+
+Additionally we can see an example of a *template helper*---`{{checkedClass}}` calls out to the `checkedClass` helper defined in a separate JavaScript file on the `todosItem` template:
+
+```js
+Template.todosItem.helpers({
+  checkedClass() {
+    return this.todo.checked && 'checked';
+  }
+});
+```
+
+In the context of a template helper, `this` is scoped to the the *data context* of the template.
```

Comment:
Yeah I write the comments as I go through the file and then push at the end, sorry. Pushed now.